src/formatDate.js

```diff
@@ -43,14 +43,53 @@
  * ) // '18.02.1997'
  *
  * @param {string} date
- * @param {string[]} fromFormat
+ * @param {string[]} string
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const string = fromFormat.join('.');
+  let year = 0;
+  const month = date.slice(string.indexOf('MM'),
+    string.indexOf('MM') + 2);
+  const day = date.slice(string.indexOf('DD'), string.indexOf('DD') + 2);
+  const separator = toFormat[3];
+  const newFormat = [];
+
+  if (fromFormat.includes('YYYY')) {
+    year = date.slice(string.indexOf('YYYY'), string.indexOf('YYYY') + 4);
+  } else {
+    year = date.slice(string.indexOf('YY'), string.indexOf('YY') + 2);
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
```

Comment:
You don't need to use a cycle for this task.
You can check indexOf year, day and month inside `fromFormat`, get value by this index from `date` and place by index from `toFormat`.